### PR TITLE
chore: Cleanup GitHub Actions

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- use ref_name only if push ref is not main (eg. we pushed a tag)
- consolidate jobs in files similar to telemetry-manager

**Related issue(s)**

### CODE REVIEW

File: .github/workflows/pr-build-image.yaml [hunk 1]
```diff
@@ -3,14 +3,29 @@ name: Build Image
...
+      - "main"
+      - "release-*"
   push:
     branches:
       - "main"
-      - "*.*.*"
-  pull_request: {}
+      - "release-*"
+
+  # while PR is open: build image with custom-components from current branch, tag: PR-XX
```
Comment (rakesh-garimella) [inline .github/workflows/pr-build-image.yaml:14]:
```suggestion
  # PR is open: build image with custom-components from current branch, tag: PR-XX
```

Comment (rakesh-garimella) [reply .github/workflows/pr-build-image.yaml:14]: Please check the alignment as well with below points

File: .github/workflows/pr-code-checks.yml [hunk 2]
```diff
@@ -10,6 +10,9 @@ on:
       - '**.md'
   workflow_dispatch:
 
+env:
```
Comment (rakesh-garimella) [inline .github/workflows/pr-code-checks.yml:13]: change name to ` PR Code checks`

File: .github/workflows/pr-code-checks.yml [hunk 3]
```diff
@@ -103,7 +106,7 @@ jobs:
         run: |
           make -j2 generate
           git diff --exit-code ':!*go.sum' || (echo 'Generated code is out of date, please run "make generate" and commit the changes in this PR.' && exit 1)
-  unittest:
+  unit-tests:
```
Comment (rakesh-garimella) [inline .github/workflows/pr-code-checks.yml:109]: Do we need `go-cache`.

File: .github/workflows/pr-code-checks.yml [hunk 4]
```diff
@@ -137,7 +140,7 @@ jobs:
 
   build:
     runs-on: ubuntu-latest
-    needs: [unittest, lint]
+    needs: [unit-tests, lint]
```
Comment (rakesh-garimella) [inline .github/workflows/pr-code-checks.yml:143]: we miss `check-coverage` step https://github.com/kyma-project/telemetry-manager/blob/main/.github/workflows/pr-code-checks.yml#L45

File: .github/workflows/tag-release.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v6
+        with:
+          distribution: goreleaser
+          version: '~> v2'
+          args: release --clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+#      - name: Release
```
Comment (rakesh-garimella) [inline .github/workflows/tag-release.yml:92]: why is this commented out

Comment (k15r) [reply .github/workflows/tag-release.yml]: gone

File: .github/workflows/pr-code-checks.yml (hunk 4 above)
Comment (k15r) [reply .github/workflows/pr-code-checks.yml:180]: done

File: .github/workflows/pr-code-checks.yml (hunk 2 above)
Comment (k15r) [reply .github/workflows/pr-code-checks.yml:13]: done

File: .github/workflows/pr-code-checks.yml (hunk 3 above)
Comment (k15r) [reply .github/workflows/pr-code-checks.yml:109]: it is not necessary but should provide a nice build time speedup

**Review by rakesh-garimella — APPROVED** [review]

**Pull request merged**